packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -703,6 +703,13 @@ class NavigatorObserver {
   /// The [Navigator] replaced `oldRoute` with `newRoute`.
   void didReplace({ Route<dynamic>? newRoute, Route<dynamic>? oldRoute }) { }
 
+  /// The top most route has changed.
+  ///
+  /// The `newRoute` is the new top most route. This can be a new route pushed
```

Comment:
I wonder if we should rename newRoute to topRoute to avoid the confusion that this is not necessarily a new-new Route?